### PR TITLE
Chore/move core to peer

### PR DESCRIPTION
**Pull request by devceline**
# Changes
- Keep track of identity key in `@web3inbox/core` valtio stores
- React to said state in the react hooks
- Update storage prefix to remove `:` per [this comment](https://github.com/WalletConnect/web3inbox-widget/pull/33#issuecomment-1752993582)
- Prepare releases (published to canaries, will update to `latest` tag if approved)

### CODE REVIEW

**Comment by bkrem** [comment]
Just a reminder: don't forget to remove the explicit `:` colon on the `customStoragePrefix` before merging/releasing with `core@2.10.2` please, otherwise this will unintentionally segment consumers storage when they update to this version 🙏

**Comment by devceline** [comment]
> Just a reminder: don't forget to remove the explicit `:` colon on the `customStoragePrefix` before merging/releasing with `core@2.10.2` please, otherwise this will unintentionally segment consumers' storage when they update to this version 🙏

Thanks for the reminder - almost slipped my mind

File: packages/core/src/client/index.ts [hunk 1]
```diff
@@ -79,8 +82,21 @@ export class Web3InboxClient {
...
 
-  public setAccount(account: string) {
-    Web3InboxClient.clientState.account = account;
+
+  /* Sets account and identity if already registered */
+  public async setAccount(account: string) {
+    // Account setting is duplicated to ensure it only gets updated once
+    // identity state is confirmed
+    if(await this.getAccountIsRegistered(account)) {
+      const identity = await this.notifyClient.identityKeys.getIdentity({ account });
```
Comment (Cali93) [inline packages/core/src/client/index.ts:91]: Could either of those throw ? How would we handle it if an error occurs ?

File: packages/core/src/client/index.ts [hunk 2]
```diff
@@ -287,6 +319,9 @@ export class Web3InboxClient {
         isLimited: true,
       });
 
+      Web3InboxClient.clientState.registration = { account: params.account, identity: regRs };
```
Comment (Cali93) [inline packages/core/src/client/index.ts:322]: Can we find a better name for `regRs` ?

File: packages/react/src/hooks/web3inboxClient.ts [hunk 3]
```diff
@@ -63,27 +63,23 @@ export const useWeb3InboxClient = () => {
...
 export const useW3iAccount = () => {
   const client = useWeb3InboxClient();
-  const [identityKey, setIdentityKey] = useState<string | null>(null);
+  const [isRegistered, setIsRegistered] = useState<boolean>(false);
 
-  const { account } = useClientState();
+  const { account, registration } = useClientState();
 
   const setAccount = useCallback(
     (account: string) => {
```
Comment (Cali93) [inline packages/react/src/hooks/web3inboxClient.ts:71]: Can we use `async/await` and add another state variable for `isRegistering` ?
```suggestion
    async (account: string) => {
```

**Review by Cali93 — APPROVED** [review]
LGTM

File: packages/core/src/client/index.ts (hunk 1 above)
Comment (devceline) [reply packages/core/src/client/index.ts:91]: `getAccountIsRegistered` can not throw.

`this.notifyClient.identityKeys.getIdentity` only throws if there is no identity, but `getAccountIsRegistered` prevents that.

File: packages/core/src/client/index.ts (hunk 2 above)
Comment (devceline) [reply packages/core/src/client/index.ts]: good idea, renamed

File: packages/react/src/hooks/web3inboxClient.ts (hunk 3 above)
Comment (devceline) [reply packages/react/src/hooks/web3inboxClient.ts]: made the function async and return the promise, and added `isRegistering` to the register method.

**Pull request merged**